fix(go/genkit): expose detailed 4xx errors in googlegenai plugin

ihan211 opened the pull request:
- Added `log.Printf` and `fmt.Errorf` in `jokesFlow` to log detailed errors.
- Used an invalid API key to test error handling.
- Updated context handling for graceful shutdown.

Tested with an invalid API key, producing:
Generate error: Error 400, Message: API key not valid...

Checklist (if applicable):
- [X] PR title is following https://www.conventionalcommits.org/en/v1.0.0/
- [X] Tested (manually, unit tested, etc.)
- [ ] Docs updated (updated docs or a docs bug required)

ihan211 committed:
fix(go/genkit): expose detailed 4xx errors in googlegenai plugin

hugoaguirre commented on go/samples/basic-gemini/main.go:
Changes should be done in the `googlegenai` plugin not in the samples. This sample can be used to validate the fix you made.

ihan211 replied:
fixed.

ihan211 committed:
fix(go/genkit): improve Google AI API key validation

- Add strict format validation to prevent silent 4xx errors
- Add detailed error handling for HTTP status codes 400, 401, 403, 429, 500
- Fix issue where invalid API keys would fail with unhelpful errors
- Document API's unusual behavior with invalid credentials

ihan211 committed:
fix(go/genkit): Improve API error handling in Google AI plugins

Remove unnecessary validation in favor of SDK's built-in validation. The Google
AI and Vertex AI plugins now rely on the SDK to validate credentials, avoiding
redundant validation steps.

Add error formatting for better error messages. API errors are now properly
formatted with clear, human-readable messages for common HTTP status codes.

Ensure errors are properly propagated to callers instead of being swallowed.
This makes debugging easier and provides better feedback to end users.

hugoaguirre commented on go/samples/basic-gemini/main.go:
You can ignore the changes in this file

ihan211 replied:
I can remove it. But the original won't take signal to exit by pressing Ctrl + C.

hugoaguirre commented on go/plugins/googlegenai/gemini.go:
```diff
@@ -327,6 +327,7 @@ func generate(
 	// Extract configuration to get the model version
 	config, err := configFromRequest(input)
 	if err != nil {
+		// This is a local error from config parsing, not an API error
```
nit: not needed (same for the rest of the comments)

ihan211 replied:
The reason I add comments is to distinguish API error and none API error.

ihan211 replied:
removed.
